swap_meet/vendor.py

```diff
@@ -1,2 +1,63 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        self.inventory.remove(item)
+        return item
+
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        self.add(their_item)
+        self.remove(my_item)
+        other_vendor.add(my_item)
+        other_vendor.remove(their_item)
+        return True
+
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False
+        their_first_item = other_vendor.inventory[0]
+        my_first_item = self.inventory[0]
+        self.swap_items(other_vendor, my_first_item, their_first_item)
+        return True
+
+    def get_by_category(self, category):
+        objects_of_category = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                objects_of_category.append(item)
+        return objects_of_category
```

Comment:
Notice this pattern of: 
```python
result_list = []
for element in source_list:
    if some_condition(element):
        result_list.append(element)
```

can be rewritten using a list comprehension as:
```python
result_list = [element for element in source_list if some_condition(element)]
```

Which here would look like:
```python
category_list = [item for item in self.inventory if item.get_category() == category]
return category_list
```

At first, this may seem more difficult to read, but comprehensions are a very common python style, so I encourage y’all to try working with them!